platforms/gke/base/use-cases/inference-ref-arch/online-inference-gpu/README.md

```diff
@@ -0,0 +1,143 @@
+
+1.  Deploy an instance of the Inference Platform reference implementation. For
+    more information about how to deploy an instance of the reference
+    architecture, see
+    [Inference Platform reference implementation](/platforms/gke/base/use-cases/inference-ref-arch/terraform/README.md)
+
+    After you deploy the reference architecture instances, continue following
+    this document.
+
+## Download the model to Cloud Storage
+
+1.  Take note of the name of the Cloud Storage bucket where the model will be
+    downloaded:
+
+    ```shell
+    terraform -chdir="${ACP_PLATFORM_USE_CASE_DIR}/terraform/cloud_storage" init \
+      && terraform -chdir="${ACP_PLATFORM_USE_CASE_DIR}/terraform/cloud_storage" output -json ira_google_storage_bucket_names
+    ```
+
+    The output might contain multiple bucket names. The name of the bucket where
+    the model will be downloaded ends with the `ira-model` suffix.
+
+1.  Initialize the configuration variables to set the name of the Cloud Storage
+    bucket where the model will be downloaded in
+    `platforms/gke/base/use-cases/inference-ref-arch/kubernetes-manifests/model-download/model-download.env`:
+
+    ```shell
+    IRA_BUCKET_NAME=<IRA_BUCKET_NAME>
+    MODEL_ID=<MODEL_ID>
+    ```
+
+    Where:
+
+    - `<IRA_BUCKET_NAME>` is the name of the Cloud Storage bucket where the
+      model will be downloaded.
+    - `MODEL_ID>` is the fully qualified model identifier.
+
+      - For Gemma, the fully qualified model identifier is:
+        `google/gemma-3-27b-it`
+      - For Llama, the fully qualified model identifier is:
+        `meta-llama/Llama-4-Scout-17B-16E-Instruct`
+
+1.  [Generate a Hugging Face token](https://huggingface.co/docs/hub/security-tokens).
+    Make sure to grant the
+    `Read access to contents of all public gated repos you can access`
+    permission to the Hugging Face token.
+
+1.  Store the Hugging Face token in
+    `platforms/gke/base/use-cases/inference-ref-arch/kubernetes-manifests/model-download/hugging-face-token.env`:
+
+    ```shell
+    HUGGING_FACE_TOKEN=<HUGGING_FACE_TOKEN>
+    ```
+
+    Where:
+
+    - `<HUGGING_FACE_TOKEN>` is the Hugging Face token.
+
+    If the
+    `platforms/gke/base/use-cases/inference-ref-arch/kubernetes-manifests/model-download/hugging-face-token.env`
+    file doesn't exist, create it.
+
+1.  Get access to the model by signing the consent agreement:
+
+    - For Gemma:
+
+      1. Access the
+         [model consent page on Kaggle.com](https://www.kaggle.com/models/google/gemma).
+
+      1. Verify consent using your Hugging Face account.
+
+      1. Accept the model terms.
+
+    - For Llama:
+
+      1. Accept the model terms on Hugging Face
+
+1.  Deploy the model downloader in the GKE cluster:
+
+    ```shell
+    kubectl apply -k platforms/gke/base/use-cases/inference-ref-arch/kubernetes-manifests/model-download
+    ```
+
+1.  Wait for the model downloader to download the model:
+
+    ```shell
+    kubectl get jobs
```

Comment:
should we add the programmatic wait here?

We should also recommend them to delete the job at the end

Reply:
> should we add the programmatic wait here?

I can do that, ok

Reply:
> We should also recommend them to delete the job at the end

I think we can add this as an optional step if they don't want to keep track of completed jobs.

Reply:
You could do something similar to what we did in the [inference use-case](https://github.com/GoogleCloudPlatform/accelerated-platforms/tree/main/use-cases/inferencing/serving/vllm/persistent-disk):

```
watch --color --interval 5 --no-title \
"kubectl --namespace=${NAMESPACE} get job/transfer-model-to-gcs | GREP_COLORS='mt=01;92' egrep --color=always -e '^' -e 'Complete'"
```

Reply:
done